### PR TITLE
Picturae import

### PR DESCRIPTION
**Pull request by delarocamcas**
files changed:
-monitoring tools: added basic function that will flag errors before import rather than at end

-added pytest at commit, updated some unittests, and fixed some issues in the unittest classes.

### CODE REVIEW

File: monitoring_tools.py [hunk 1]
```diff
@@ -21,6 +23,12 @@ def clear_txt(self):
         with open(self.path, 'w') as file:
             pass
 
+    def check_config_present(self):
```
Comment (delarocamcas) [inline monitoring_tools.py:26]: function to check whether the correct attributes are in the config file.

File: sql_csv_utils.py [hunk 2]
```diff
@@ -16,7 +16,16 @@ def __init__(self, config, logging_level=logging.INFO):
         self.specify_db_connection = SpecifyDb(db_config_class=self.config)
         self.logger = logging.getLogger("SqlCsvTools")
         self.logger.setLevel(logging_level)
-        self.specify_db_connection.connect()
+        self.check_db_connection()
+
+    def check_db_connection(self):
```
Comment (delarocamcas) [inline sql_csv_utils.py:21]: checks whether db connection is function in sql_csv_tools at beginning of class init

File: tests/pic_csv_test_class.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 from picturae_csv_create import CsvCreatePicturae
 import picturae_config
 
-class TestCsvCreatePicturae(CsvCreatePicturae):
+class AltCsvCreatePicturae(CsvCreatePicturae):
```
Comment (delarocamcas) [inline tests/pic_csv_test_class.py:6]: changed name to Alt, as pytest was trying to initialize initialize it as a test class

File: tests/test_sql_tools.py [hunk 4]
```diff
@@ -69,60 +67,89 @@ def test_remove_two_index(self):
         for term in test_values:
             self.assertTrue(not pd.isna(term) and term not in [None, np.nan, ''])
 
-    def test_create_locality(self):
-        """testing create_locality function by
-           recreating insert protocol for locality table, but with sqlite DB"""
+    def test_insert_table_record(self):
```
Comment (delarocamcas) [inline tests/test_sql_tools.py:70]: newer version of function required a collecting event id, so had to switch to manually creating the insert statement to test the insert function.

File: sql_csv_utils.py [hunk 5]
```diff
@@ -16,7 +16,16 @@ def __init__(self, config, logging_level=logging.INFO):
...
+        """checking whether database connection is functional"""
+        try:
+            self.specify_db_connection.connect()
+            self.logger.info("sql_csv_tools connection established")
+        except Exception as e:
+            raise ValueError(f"Database connection error: {e}")
```
Comment (foozleface) [inline sql_csv_utils.py:27]: You can create a custom exception by inherting from the "Exception" class. "valueException" is a poor choice; I'd say create a custom "DatabaseConnectionError". there are examples of this in the code.

**Review by foozleface — CHANGES_REQUESTED** [review]

File: sql_csv_utils.py (hunk 5 above)
Comment (delarocamcas) [reply sql_csv_utils.py:27]: ok, will do

**Review by foozleface — APPROVED** [review]

**Pull request merged**